src/controllers/controllerengine.cpp

```diff
@@ -486,21 +483,19 @@ bool ControllerEngine::execute(QString function, const QByteArray data) {
 
 /**-------- ------------------------------------------------------
    Purpose: Evaluate & call a script function
-   Input:   Function name, ponter to data buffer, length of buffer
+   Input:   Function name, pointer to data buffer, length of buffer
    Output:  false if an invalid function or an exception
    -------- ------------------------------------------------------ */
 bool ControllerEngine::execute(QScriptValue function, const QByteArray data) {
-    if (m_pEngine == NULL) {
+    if (!function.isFunction()) {
         return false;
     }
-
-    if (checkException())
-        return false;
-    if (!function.isFunction())
+    if (checkException()) {
         return false;
+    }
 
     QScriptValueList args;
-    args << QScriptValue(m_pBaClass->newInstance(data));
+    args << m_pBaClass->newInstance(data);
```

Comment:
How can you remove the QScriptValue cast? Anything passing to script must be a QScriptValue if I recall correctly.